contracts/src/libraries/PolynomialEval.sol

```diff
@@ -195,19 +195,20 @@ library PolynomialEval {
         uint256[] memory pi,
         uint256 zeta,
         uint256 vanishingPolyEval
-    ) internal view returns (uint256 res) {
-        uint256 length = pi.length;
+    ) internal view returns (uint256 res, uint256 evalLagrangeOne) {
+        // TODO document better that pi.length=8
+
         uint256 p = BN254.R_MOD;
 
         if (vanishingPolyEval == 0) {
             uint256 group = 1;
-            for (uint256 i = 0; i < length; i++) {
+            for (uint256 i = 0; i < 8; i++) {
                 if (zeta == group) {
-                    return pi[i];
+                    return (pi[i], 1);
                 }
                 group = mulmod(group, self.groupGen, p);
             }
-            return 0;
+            return (0, 0);
```

Comment:
this is wrong logic btw, when `vanishingPolyEval == 0`, the zeta could be one of the domain element: if it's the first domain element, then `lagrange_1_eval = 1`

Reply:
Thanks. Some tests still fail though.